packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -1165,16 +1168,37 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
       ...this.initialOptions,
       ...options
     };
-    try {
-      await this.mongoClient.close();
-      // eslint-disable-next-line no-empty
-    } catch {}
     const mc = await MongoClient.connect(
       this.uri as string,
       clientOptions
     );
+    try {
```

Comment:
This was a bug where if the reconnection failed for whatever reason, we had already closed the old client so could not restore the connection